front/lib/workspace_usage.ts

```diff
@@ -104,6 +150,301 @@ export async function unsafeGetUsageData(
+        .lastMessageSent,
+      activeDaysCount: (result as unknown as { activeDaysCount: number })
+        .activeDaysCount,
+    };
+  });
+  if (!userUsage.length) {
+    return "No data available for the selected period.";
+  }
+  const csvHeader = Object.keys(userUsage[0]).join(",") + "\n";
+  const csvContent = userUsage
+    .map((row) => Object.values(row).join(","))
+    .join("\n");
+  return csvHeader + csvContent;
+}
+
+export async function getBuildersUsage(
+  workspaceId: string,
+  startDate: Date,
+  endDate: Date
+): Promise<string> {
+  const workspace = await Workspace.findOne({
+    where: { sId: workspaceId },
+  });
+  if (!workspace) {
+    throw new Error(`Workspace not found for sId: ${workspaceId}`);
+  }
+  const wId = workspace.id;
+  const agentConfigurations = await AgentConfiguration.findAll({
+    attributes: [
+      [
+        Sequelize.fn("COUNT", Sequelize.col("agent_configuration.sId")),
+        "agentsEditionsCount",
+      ],
+      "user.email",
+      "user.firstName",
+      "user.lastName",
+      [
+        Sequelize.fn(
+          "COUNT",
+          Sequelize.literal('DISTINCT "agent_configuration"."sId"')
+        ),
+        "distinctAgentsEditionsCount",
+      ],
+    ],
+    where: {
+      workspaceId: wId,
+      createdAt: {
+        [Op.gt]: startDate,
+        [Op.lt]: endDate,
+      },
+    },
+    include: [
+      {
+        model: User,
+        as: "user",
+        attributes: [],
+        required: true,
+      },
+    ],
+    raw: true,
+    group: ["authorId", "user.email", "user.firstName", "user.lastName"],
+  });
+  const buildersUsage: builderUsage[] = agentConfigurations.map((result) => {
+    return {
+      userFirstName: (result as unknown as { firstName: string }).firstName,
+      userLastName: (result as unknown as { lastName: string }).lastName,
+      userEmail: (result as unknown as { email: string }).email,
+      agentsEditionsCount: (
+        result as unknown as { agentsEditionsCount: number }
+      ).agentsEditionsCount,
+      distinctAgentsEditionsCount: (
+        result as unknown as { distinctAgentsEditionsCount: number }
+      ).distinctAgentsEditionsCount,
+    };
+  });
+  if (!buildersUsage.length) {
+    return "No data available for the selected period.";
+  }
+  const csvHeader = Object.keys(buildersUsage[0]).join(",") + "\n";
+  const csvContent = buildersUsage
+    .map((row) => Object.values(row).join(","))
+    .join("\n");
+  return csvHeader + csvContent;
+}
+
+export async function getAgentUsage(
+  workspaceId: string,
+  startDate: Date,
+  endDate: Date
+): Promise<string> {
+  const workspace = await Workspace.findOne({
+    where: { sId: workspaceId },
+  });
+  if (!workspace) {
+    throw new Error(`Workspace not found for sId: ${workspaceId}`);
+  }
+  const wId = workspace.id;
+  const mentions = await frontSequelize.query<AgentUsageQueryResult>(
+    `
+    SELECT
```

Comment:
nit: IMHO big queries like that make the whole file harder to read (+ psychological blow when seeing it :) )

would suggest creating separate .sql files for those queries; this way @thib-martin could even edit them directly (and make a comment as to what they are)
and we load them as top level constant
asking another eng for their opinion here e.g. @flvndvd

Reply:
💯 Sounds good to me!